### PR TITLE
Forgot password

### PR DESCRIPTION
**Pull request by bvangennep**

### CODE REVIEW

File: controllers/ApiAuthController.php [hunk 1]
```diff
@@ -16,30 +16,36 @@
...
         }
+    }
 
-        try{
+    /**
+     * Authenticate action.
+     */
+    public function actionAuthenticate()
+    {
+
```
Comment (boboldehampsink) [inline controllers/ApiAuthController.php]: Please remove this unused space

File: controllers/ApiAuthController.php [hunk 2]
```diff
@@ -52,7 +58,31 @@ public function actionAuthenticate()
...
+            $this->returnErrorJson($e->getMessage());
+        }
+    }
+
+    /**
+     * Forgot password action
+     */
+    public function actionResetPassword()
+    {
+
```
Comment (boboldehampsink) [inline controllers/ApiAuthController.php]: Please remove this unused space

File: controllers/ApiAuthController.php [hunk 3]
```diff
@@ -52,7 +58,31 @@ public function actionAuthenticate()
...
+            $username = craft()->request->getRequiredPost('username');
+            $user = craft()->users->getUserByUsernameOrEmail($username);
+
+            if ($user) {
+                craft()->users->sendPasswordResetEmail($user);
+            }
+
+            $this->returnJson(array('message' => Craft::t('Email has been sent if address exists')));
+
```
Comment (boboldehampsink) [inline controllers/ApiAuthController.php:81]: Please remove this unused space

File: services/ApiAuthService.php [hunk 4]
```diff
@@ -80,11 +80,22 @@ protected function getNewUserKeyModel()
...
+
+    /**
+     * @codeCoverageIgnore
+     *
+     * @param string $key
+     * @param string $value
+     */
+    protected function setHeader($key, $value)
+    {
+        HeaderHelper::setHeader("$key: $value");
```
Comment (boboldehampsink) [inline services/ApiAuthService.php]: Please don't write variables in quotes

File: services/ApiAuthService.php [hunk 5]
```diff
@@ -80,11 +80,22 @@ protected function getNewUserKeyModel()
...
+     * @param string $key
      * @return ApiAuth_UserKeyModel
      */
     protected function getUserKeyModelByKey($key)
     {
         return ApiAuth_UserKeyRecord::model()->findByAttributes(array('key' => $key));
     }
+
+    /**
+     * @codeCoverageIgnore
```
Comment (boboldehampsink) [inline services/ApiAuthService.php:92]: Why ignore this?

**Comment by boboldehampsink** [comment]
Some little style issues, but looking good :+1:

File: services/ApiAuthService.php (hunk 5 above)
Comment (bvangennep) [inline services/ApiAuthService.php:92]: Because use of static functions cannot be tested

**Pull request merged**